context/app/notebooks.py

```diff
@@ -0,0 +1,128 @@
+import json
+
+from nbformat.v4 import new_code_cell
+import black
+
```

Comment:
A little difficult to read with the formatting as is, is there a way to get more of this formatted as python?

Reply:
Referring to the f strings in the functions in `notebooks.py`.

Reply:
I looked a little bit at using python introspection, so we could just define functions, and then inject the definitions of the functions. Didn't go that route because:
- The introspection utils would return the whole function body, and then we'd have to strip the first line.
- Unlike javascript, python won't let you define a multi-line function as an expression... so we'd be defining the functions in one place, and then using them another place.
- The introspection just gets kind of gnarly: I'd worry about debugging it.

So: Not really happy with the big strings, but the alternative felt like it was introducing a lot of complexity which would have been an additional burden.

Reply:
Another idea: We can check in an actual notebook to store the cell definitions. Not sure whether the variable parts will just be prepended, or if we can use a template syntax without introducing too much escaping.